### PR TITLE
DS-235 Remove ie11 polyfills and unused packages

### PR DESCRIPTION
**Pull request by danielamorse**
## Jira

https://pegadigitalit.atlassian.net/browse/DS-235

## Summary

Remove IE11 polyfills and unused packages.

## Details
Removed IE11 polyfills from:
- `@bolt/polyfills` (provides only the custom-elements polyfill required by Edge < 79)
- `@bolt/core-v3.x`
- `@bolt/micro-journeys`
- `@bolt/analytics-autotrack`

Specific polyfill dependencies removed:
- `@ungap/weakset`
- `@webcomponents/shadycss`
- `@webcomponents/shadydom`
- `@webcomponents/template`
- `@webcomponents/url`
- `@webcomponents/webcomponentsjs`
- `core-js`
- `custom-event-polyfill`
- `document-register-element`
- `es6-promise`
- `get-own-property-symbols`
- `promise-polyfill`
- `element-closest`
- `mdn-polyfills`
- `smoothscroll-polyfill`
- `whatwg-fetch`
- `regenerator-runtime/runtime`

Deleted the following unused packages:
- `@bolt/core` (aka 'core-v2.x')
- `@bolt/components-critical-fonts`
- `@bolt/components-critical-css`

> Note: these details, along with the other `v3.0` PR details, will be compiled in the release notes and upgrade guide.

## How to test

There are 200+ changed files. The best way to understand the code changes is to look at each commit, where related changes are grouped together in a way that's more digestible than viewing the full diff.

- Review files changed.
- Build is passing
- No regressions (including Edge 18)

## Release notes

Rename any dependencies on `@bolt/core` to `@bolt/core-v3.x`. Remove any use of `@bolt/components-critical-fonts` and `@bolt/components-critical-css`.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,16 +1,13 @@
...
+  "include": ["packages", "scripts"],
   "exclude": [
     "dist",
     "www",
     "**/vendor/**",
     "packages/generators/yeoman-create-component/generators/component",
     "packages/components/bolt-icons/",
     "packages/components/bolt-icon",
-    "packages/website-ui/uikit-workshop",
+    "packages/website-ui/uikit-workshop"
```
Comment (remydenton) [inline tsconfig.json:10]: Nitpick, don't we want to keep the trailing comma here?

File: packages/base-polyfills/index.js [hunk 2]
```diff
@@ -9,62 +9,7 @@ subject to an additional IP rights grant found at http://polymer.github.io/PATEN
...
-      i = matches.length;
-      while (--i >= 0 && matches.item(i) !== el) {}
-    } while (i < 0 && (el = el.parentElement));
-    return el;
-  };
-}
-
-if (window.NodeList && !NodeList.prototype.forEach) {
-  NodeList.prototype.forEach = Array.prototype.forEach;
-}
```
Comment (remydenton) [inline packages/base-polyfills/index.js:70]: So much polyfill cruft removed 😍 !

File: packages/core-v3.x/styles/02-tools/tools-themes/_tools.themes.scss [hunk 3]
```diff
@@ -85,9 +85,9 @@ $PSEUDO_CLASSES: (
 
 /**
   * Helper functions for applying global color swatches stored in Sass Maps. See @settings-themes/index.scss
-  * @param {map|...params} $props
-  *   - Map values are printed json encoded straight to bolt-themify to be processed by webpack plugin postcss-themify
-  *   - Other args generate a bolt-themify json-encoded map according to theme settings in `settings-themes/index.scss`
+  * @param {...params} $props
+  *   - Color name 
+  *   - Opacity
```
Comment (remydenton) [inline packages/core-v3.x/styles/02-tools/tools-themes/_tools.themes.scss:90]: So much simpler 😍 !

File: packages/build-tools/tasks/api-tasks/bolt-vrt-urls.js [hunk 4]
```diff
@@ -40,9 +40,6 @@ async function generateVrtUrls() {
     '@bolt/components-radio-switch',
     '@bolt/components-animate',
     '@bolt/components-card',
-    '@bolt/components-critical-css-vars',
-    '@bolt/components-critical-css',
-    '@bolt/components-critical-fonts',
```
Comment (remydenton) [inline packages/build-tools/tasks/api-tasks/bolt-vrt-urls.js:45]: I know these are currently inlined on the page in Drupal.  Does it make sense to outline the update steps (i.e. remove those) in this PR, or is that coming somewhere else?

**Review by remydenton — APPROVED** [review]
Looks great.  Looking at this PR makes me feel like I've just taken a shower or something 🛀 .

File: tsconfig.json (hunk 1 above)
Comment (danielamorse) [reply tsconfig.json:10]: Ah, I thought so too but realized it's working as expected for JSON, not JS.

File: packages/build-tools/tasks/api-tasks/bolt-vrt-urls.js (hunk 4 above)
Comment (danielamorse) [reply packages/build-tools/tasks/api-tasks/bolt-vrt-urls.js:45]: Definitely, there's a guide here: https://github.com/boltdesignsystem/bolt/blob/master/docs-site/src/pages/docs/50-guides/60-migrating-to-3.x.md

These items^ are already included in the guide, but I'll give it a once over before you test it in Drupal.

**Review by colbytcook — APPROVED** [review]

File: tsconfig.json (hunk 1 above)
Comment (remydenton) [reply tsconfig.json:10]: Ah right, nevermind me!

**Pull request merged**